st3/sublime_lib/fancy_settings.py

```diff
@@ -158,12 +160,24 @@ class NamedFancySettings(FancySettings):
     file.
     """
 
-    def __init__(self, name, defaults={}):
+    def __init__(self, name):
         """Return a new NamedFancySettings corresponding to the given name."""
 
-        super().__init__(sublime.load_settings(name), defaults)
+        super().__init__(sublime.load_settings(name))
         self.name = name
 
     def save(self):
         """Flushes any in-memory changes to the named settings object to disk."""
         sublime.save_settings(self.name)
+
+
+class DefaultFancySettings(FancySettings):
+    def __init__(self, settings, defaults):
+        """
+        Return a new FancySettings wrapping a given Settings object *settings*.
+        """
+        super().__init__(settings)
+        self.defaults = defaults
+
+    def __missing__(self, key):
+        return self.setdefault(key, self.defaults[key])
```

Comment:
Do we actually need `setdefault`? What's the motivation behind it over `get`?

For defaultdicts it makes sense since the default factory function would, say, create a new list so that you can do stuff like 
```py
ddict = defaultdict(list)
ddict['nonexistant'] += [1]
ddict['nonexistant'] += [2]
ddict['nonexistant'] == [1, 2]
```
, but we don't need this here since we return existing references.

Either way, if `setdefault` was to be used, its side-effect should be documented.

Reply:
I've modified `__missing__` to work exactly the same as in `defaultdict`. When `__missing__` is called, it will unconditionally set the value to the default (even if it had another value). This shouldn't matter much because users probably shouldn't manually call `__missing__`, but at least it's consistent.

Using `get` would make sense as well, but all else being equal it's probably best to stick with the existing convention.

Documentation has been updated.